diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
index f044d22d4d..a78b9aa1e9 100644
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -203,7 +203,7 @@
     if(wxBUILD_SHARED)
-        set(lib_suffix "_dll")
+        # set(lib_suffix "_dll")
     else()
-        set(lib_suffix "_lib")
+        # set(lib_suffix "_lib")
     endif()
-    set(wxPLATFORM_LIB_DIR "${wxCOMPILER_PREFIX}${wxARCH_SUFFIX}${lib_suffix}")
+    # set(wxPLATFORM_LIB_DIR "${wxCOMPILER_PREFIX}${wxARCH_SUFFIX}${lib_suffix}")
     set(wxINSTALL_INCLUDE_DIR "${include_dir}")
