diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
index 5447d33..f5440b4 100644
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -693,3 +693,5 @@
     if(wxUSE_SOUND AND wxUSE_LIBSDL AND UNIX AND NOT APPLE)
-        find_package(SDL2)
+        find_package(SDL2 CONFIG REQUIRED)
+        set(SDL2_INCLUDE_DIR "" CACHE INTERNAL "")
+        set(SDL2_LIBRARY SDL2::SDL2 CACHE INTERNAL "")
         if(NOT SDL2_FOUND)
diff --git a/build/cmake/wxWidgetsConfig.cmake.in b/build/cmake/wxWidgetsConfig.cmake.in
--- a/build/cmake/wxWidgetsConfig.cmake.in
+++ b/build/cmake/wxWidgetsConfig.cmake.in
@@ -179,2 +179,5 @@ include(CMakeFindDependencyMacro)
 
+if("@wxUSE_LIBSDL@")
+    find_dependency(SDL2 CONFIG)
+endif()
 if(TARGET wx::wxnet AND @wxUSE_WEBREQUEST_CURL@)
